src/main/java/hu/dpc/phee/operator/util/BatchFormatToTransferMapper.java

```diff
@@ -55,10 +56,19 @@ public static Transfer updateTransferUsingBatchDetails(Transfer transfer, Batch
                 transfer.setStatus(TransferStatus.COMPLETED);
             } else {
                 if (batch.getTotalTransactions() != null && batch.getCompleted() != null) {
-                    if (batch.getTotalTransactions() == batch.getCompleted()) {
+                    if (Objects.equals(batch.getTotalTransactions(), batch.getCompleted())) {
                         transfer.setStatus(TransferStatus.COMPLETED);
                     }
                 }
+                if (batch.getCompleted().longValue() == batch.getTotalTransactions().longValue()) {
```

Comment:
batch.getTotalTransactions().longValue() can cause a null pointer as per the above code. there is an explicit null check added there

Reply:
add a null safe condition

Reply:
done , please merge this.